libs/nest/aws/src/lib/s3.service.ts

```
@@ -158,6 +160,28 @@ export class S3Service {
     }
   }
 
+  public async getFileTags(
+    BucketKeyPairOrFilename: BucketKeyPair | string,
+  ): Promise<Tag[]> {
+    const { bucket, key } = this.getBucketKey(BucketKeyPairOrFilename)
+    
+    try {
+      const command = new GetObjectTaggingCommand({
+        Bucket: bucket,
+        Key: key,
+      })
+      const results = await this.s3Client.send(command)
+      
+      return results?.TagSet ?? []
+    } catch (error) {
+      this.logger.error(
+        `Error occurred while fetching file tags for key: ${key} in S3 bucket: ${bucket}`,
+        error,
+      )
+    }
+    return []
```

Should almost never swallow unexpected errors. Errors from user input should only be logged as warnings (eg bad url?). 

Should looking up tags for a file which does not exist return an empty array or throw an error? I prefer throwing an error so the client knows that something happened, instead of assuming the file has no tags.

---

Ah good point, getting ahead of myself a bit here 😅 Fixed.